guidance/library/_regex.py

```diff
@@ -124,14 +123,8 @@ def MAX_REPEAT(
         if isinstance(high, constants._NamedIntConstant):
             if high != constants.MAXREPEAT:
                 raise UnsupportedRegexError(f"Unsupported high value in range: {high}")
-            if low == 0:
-                # kleene star
-                return zero_or_more(transformed_arg)
-            if low > 0:
-                return Join([transformed_arg] * low + [zero_or_more(transformed_arg)])
-        return Join(
-            [transformed_arg] * low + [optional(transformed_arg)] * (high - low)
-        )
+            return sequence(transformed_arg, min_length=low)
```

Comment:
I assume this is what you were meaning by regex @hudson-ai ?

Reply:
Yes, thank you!